src/System.ServiceProcess.ServiceController/src/System/ServiceProcess/ServiceBase.cs

```diff
@@ -659,6 +646,14 @@ public static void Run(ServiceBase[] services)
             }
             finally
             {
+                // Free the pointer to the name of the service on the unmanaged heap.
+                // We don't need to free the last element in the unmanaged array since the last element have null values (denotes the end of the table).
+                for (int i = 0; i < services.Length; i++)
```

Comment:
```suggestion
                for (int i = 0; i < entries.Length; i++)
```

Since this loop is only touching `entries`, I would make the for-loop to over entries. It is actually tiny bit better - the JIT elides the null check for canonical loops like `for (int i = 0; i < a.Length; i++) ... do stuff with a[i]`.